src/Dfe.Spi.EntitySquasher/Dfe.Spi.EntitySquasher.FunctionApp/Functions/GetSquashedEntity.cs

```diff
@@ -63,30 +60,102 @@ public class GetSquashedEntity
                 throw new ArgumentNullException(nameof(httpRequest));
             }
 
-            string getSquashedEntityRequestStr = null;
-            using (StreamReader streamReader = new StreamReader(httpRequest.Body))
+            // Set the context of the logger.
+            this.loggerWrapper.SetContext(httpRequest.Headers);
+
+            GetSquashedEntityRequest getSquashedEntityRequest = null;
+            try
             {
-                getSquashedEntityRequestStr = streamReader.ReadToEnd();
+                getSquashedEntityRequest = this.ParseRequest(httpRequest);
+            }
+            catch (JsonReaderException)
+            {
+                // Do nothing - getSquashedEntityRequest will just end up null.
             }
 
-            GetSquashedEntityRequest getSquashedEntityRequest =
-                JsonConvert.DeserializeObject<GetSquashedEntityRequest>(
-                    getSquashedEntityRequestStr);
+            if (getSquashedEntityRequest != null)
+            {
+                this.loggerWrapper.Debug(
+                    $"Invoking {nameof(IGetSquashedEntityProcessor)}...");
+
+                toReturn = await this.ProcessWellFormedRequestAsync(
+                    getSquashedEntityRequest)
+                    .ConfigureAwait(false);
+            }
+            else
+            {
+                int statusCode = StatusCodes.Status400BadRequest;
+
+                this.loggerWrapper.Warning(
```

Comment:
This should probably not be a warning should it? If you get this in an alert, what are you going to do?

I would expect it to be info and the fact that a non-success code being returned be handled in the caller.

Reply:
I guess it's just how you view error levels. I try to be consistant with my own definitions. These are:

- **Debug**: everything - anything that might be useful in getting you out of a pinch in a live situation to figure out why a bug is happening;
- **Info**: an abbreviated, less verbose version of the above, but containing key information to ensure smooth running;
- **Warn**: Anything you might want to know about from an alerting perspecitve, but isn't an all out error - for example, you might want to know that someone's posting empty JSON objects to the API, as this isn't a usual thing - it could be an attack or similar. It's not an error, and doesn't affect the smooth running of the system;
- **Error**: Anything that shouldn't happen, but isn't fatal;
- **Fatal**: This shouldn't happen, and it ends the current process.

Reply:
So I largely agree with your definitions, with 2 exceptions:

1. I am not sure bad requests are warnings (as I said). They will generally be problems with the calling party which we do not care about. As you say, they do not affect the smooth running of our system and we are not going to pro-actively do anything about them. If the service was being attacked then we would likely find that through other telemetry (such as a general increase in 4xx/5xx responses in AI)

2. I am not sure debug should be on in Live. Feels like this level should be for more local purposes. But this I am less worried about, as logging throughput it unlikely to be an issue in out use case

Reply:
Based on conversation, have changed this.